packages/dynamite/lib/src/openapi_builder.dart

```diff
@@ -1242,8 +1269,14 @@ TypeResult resolveObject(
           final sortedParameterKeys = schema.properties!.keys.toList()
             ..sort(
               (final a, final b) => sortRequiredElements(
-                (schema.required ?? []).contains(a) && schema.properties![a]!.default_ == null,
-                (schema.required ?? []).contains(b) && schema.properties![b]!.default_ == null,
+                _isDartParameterRequired(
+                  (schema.required ?? []).contains(a),
```

Comment:
the receiver can take a nullable value so `schema.required?.contains(a)` might be cleaner.

Also everywhere else this is done